iree/vm/list.c

```diff
@@ -686,3 +686,124 @@ iree_status_t iree_vm_list_register_types(void) {
   iree_vm_list_descriptor.type_name = iree_make_cstring_view("vm.list");
   return iree_vm_ref_register_type(&iree_vm_list_descriptor);
 }
+
+#define std_swap(a, b) \
+  do {                 \
+    typeof(a) t = a;   \
```

Comment:
there's no typeof in C and this is compiler-specific behavior (won't work in msvc, for example); a simple way that doesn't require magic is to just take the type as an argument:
```c
#define iree_swap(type, a, b) ...
iree_swap(iree_host_size_t, src->foo, dst->foo);
```

(this could be handled with [_Generic](https://en.cppreference.com/w/c/language/generic) and may be useful to add to iree/base/, but it's probably not worth the complexity for this one use today :)

Reply:
I see! Based on the input type and return the right swap variation:
```c
#define iree_swap(a, b) _Generic((a), \
              long double: iree_swap_f64, \
                    float: iree_swap_f32, ...  \
)(a, b)
```
Looks like we will need: i8/i16/i32/i64/f16/f32/f64 ?
I can try this as well if we decide to do (hope i didn't oversimplify it)